keras_cv/models/adain.py

```diff
@@ -0,0 +1,228 @@
+    layer_names = ["block1_conv1", "block2_conv1", "block3_conv1", "block4_conv1"]
+    outputs = [vgg19.get_layer(name).output for name in layer_names]
+    mini_vgg19 = keras.Model(vgg19.input, outputs)
+
+    inputs = layers.Input([*IMAGE_SIZE, 3])
+    mini_vgg19_out = mini_vgg19(inputs)
+    return keras.Model(inputs, mini_vgg19_out, name="loss_net")
+
+
+def get_mean_std(x):
+    """Computes the mean and standard deviation of a tensor."""
+    axes = [1, 2]
+    mean, variance = tf.nn.moments(x, axes=axes, keepdims=True)
+    standard_deviation = tf.sqrt(variance + 1e-5)
+    return mean, standard_deviation
+
+
+class AdaIN(layers.Layer):
+    def __init__(self, name='AdaIN', **kwargs):
+        super(AdaIN, self).__init__(name=name, **kwargs)
+
+    def call(self, inputs):
+        """Computes the AdaIn feature map.
+
+        Args:
+            style: The style feature map.
+            content: The content feature map.
+
+        Returns:
+            The AdaIN feature map.
+        """
+        style, content = inputs
+        content_mean, content_std = get_mean_std(content)
+        style_mean, style_std = get_mean_std(style)
+        t = style_std * (content - content_mean) / content_std + style_mean
+        return t
+
+
+class AdaINModel(keras.Model):
+    def __init__(self, image_size=(None, None), style_weight=4.0, **kwargs):
+        super().__init__(**kwargs)
+        self.image_size = image_size
+        self.encoder = get_mini_vgg19_encoder(image_size)
+        self.decoder = get_adain_decoder()
+        self.adain_layer = AdaIN()
+        self.loss_net = get_loss_net(image_size)
+        self.style_weight = style_weight
+
+    def produce_functional_model():
+        style_input = layers.Input([*self.image_size, 3])
+        style_encoded = self.encoder(style_input)
+        content_input = layers.Input([*self.image_size, 3])
+        content_encoded = self.encoder(content_input)
+
+        adain_feature_map = self.adain_layer((style_encoded, content_encoded))
+        content_decoded = self.decoder(adain_feature_map)
+        
+        return keras.Model([style_input, content_input], content_decoded, name="adain_style_transfer")
+
+
+    def compile(self, optimizer, loss_fn):
+        super().compile()
+        self.optimizer = optimizer
+        self.loss_fn = loss_fn
+        self.style_loss_tracker = keras.metrics.Mean(name="style_loss")
+        self.content_loss_tracker = keras.metrics.Mean(name="content_loss")
+        self.total_loss_tracker = keras.metrics.Mean(name="total_loss")
+        
+    def call(self, inputs):
+        style, content = inputs
+        # Encode the style and content image.
+        style_encoded = self.encoder(style)
+        content_encoded = self.encoder(content)
+
+        # Compute the AdaIN target feature maps.
+        t = self.adain_layer((style_encoded, content_encoded))
+
+        # Generate the neural style transferred image.
+        return self.decoder(t), t
```

Comment:
Let's actually do this in the decoder - that way we don't have to think about it, and the decoder goes from the learned space `R_n` => image in range 0, 255.

So in the production of the decoder, lets add a multiply by 0, 255 at the end of the network and clip to 0, 255.  This will probably required you to migrate to a functional model - otherwise SavedModel will break for the Sequential.

you'll basically just iterate the loops you have defined, then add:

```
x = tf.clip_by_value(x * 255., 0., 255.0)
```